### PR TITLE
How to build a quiz app with Vanilla JavaScript and Tailwind CSS

### LINKED ISSUE #4083: How to build a Quiz App with Vanilla JavaScript and Tailwind CSS
#### Proposed title of article
How to build a Quiz App with Vanilla JavaScript and Tailwind CSS

#### Proposed article introduction
This article will show you how to build a quiz app with HTML, CSS and JavaScript and Tailwind CSS. The app will contain several questions and answers. Each page will have a question and four options, and if a user gets the answer correct, he moves to the next question. If not, he fails. It is a must-do javascript project.

#### Key takeaways 
1. Readers will learn the functions involved in creating a quiz app with just Vanilla JavaScript.
2. Readers will get a basic understanding of creating any web application with JavaScript.
3. Readers will learn how to integrate Tailwind CSS in their JavaScript web applications

#### Article quality
This article will be unique because I will be integrating Tailwind CSS into a JavaScript quiz app which is most likely going to improve its SEO. It is also a great project for JavaScript developers as the project touches different areas of JavaScript. 

#### References
N/A

### PR DESCRIPTION
**Pull request by Nomzy-kush**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This fixes #4083

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
@Nomzy-kush, you must add this in the initial comment as unfortunately, it does not work in subsequent ones.

**Comment by Nomzy-kush** [comment]
> @Nomzy-kush, you must add this in the initial comment as unfortunately, it does not work in subsequent ones.

Alright, I have done that

**Comment by hectorkambow** [comment]
@M0nicah 
Can we get a status update on this PR?

**Comment by hectorkambow** [comment]
Pinged reviewer via Slack

File: content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+The code above represents the styling for the answer buttons and their responsiveness to clicking when the user chooses an answer.
+
+Note that we also set a `hide` class to `display: none`, which we will use in JavaScript.
+
+This is how the root page of the app looks like at this point:
+
+![quiz-app]/engineering-education/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/quiz-app.jpg)
+
+### Now the JavaScript
```
Comment (M0nicah) [inline content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md:204]: Try to use another heading. This does not seem to fit here.

File: content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+First, create a `script.js` file in your app directory. In there, you have to call the HTML elements you will work on into your JavaScript with the code below:
+```js
+const startButton = document.querySelector('#start-btn')
+const nextButton = document.querySelector('#next-btn')
+const questionContainerElement = document.querySelector('#question-container')
+const questionElement = document.querySelector('#question')
+const answerButtonsElement = document.querySelector('#answer-buttons')
+```
+Then, you have to create an array that will contain your quiz questions and their corresponding answer options.
```
Comment (M0nicah) [inline content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md:214]: You have used `you have to` many times at this point. Try to use another phrase?

File: content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+```js
+const clearStatusClass = (element) => {
+    element.classList.remove('correct')
+    element.classList.remove('wrong')
+}
+```
+
+Below is a video of our quiz application in action:
+
+<iframe width="727" height="409" src="https://www.youtube.com/embed/ENdP7aFVFpY" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
```
Comment (M0nicah) [inline content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md:403]: if you want to add a video for your article, you can screen record it then add it to your markdown as a `.gif`. The `<iframe>` doesn't seem to have worked.

File: content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+  }
+}
+```
+
+Lastly, run the command below in your terminal, and then Tailwind CSS will be set up in your application.
+
+`npm run tw:build`
+
+You can create your `index.html` in your `public/css` folder. Your `style.css` will be in your `src/css` folder. The app directories may look like this:
+
```
Comment (M0nicah) [inline content/articles/how-to-build-a-quiz-app-with-vanilla-javascript-and-tailwind-css/index.md:86]: Something is missing here🤔

**Pull request merged**